ryanoasis/nerd-fonts
# Pull 'Mono' to front in names and other naming changes

This does a lot more than initially planned.

### Main features
* `Complete` is dropped from the names (we have no incomplete fonts anymore anyhow)
* `Windows Compatible` fonts are dropped (all fonts are now 'windows compatible')
* The weights are now always placed last in the name (as 'standard' expects)
* The `Nerd Font Mono` is treated as one connected name part
* The blanks are dropped from the filenames

### Additional features
* Variant `Nerd Font Propo` is now properly named and included in patched-fonts
* Each `font-patcher` run creates a logfile with debug info
* All fonts are created with the new naming engine (`--makegroups`)
* The `--makegroups` option gets a 'secret' parameter that controls font name shortening
* New options `--dry` and `--debug` for `font-patcher` (drop `--windows` and `--also-windows`)
* Envvar `NERDFONTS` can be used to pass additional flags on `gotta-patch-em` runs

### Name changes
* AurulentSansMono -> AurulentSansM
* BigBlueTerminal -> BigBlueTerm
* DejaVuSansMono -> DejaVuSansM
* FantasqueSansMono -> FantasqueSansM
* FiraCode Retina -> FiraCode Ret
* iM-WritingMonoS -> iM-WritingMono
* iM-WritingDuoS -> iM-WritingDuo
* iM-WritingQuattroS -> iM-WritingQuat
* Terminess TTF -> Terminess

## How should this be manually tested?

One way is to run

```
$ NERDFONTS='--dry --debug' ./gotta-patch-em-all-font-patcher\!.sh -c /Terminus | \
      grep 'CRIT|-<|ERROR|WARN|mode|===='
```

and examine the output. Or

```
$ ./gotta-patch-em-all-font-patcher\!.sh -cj
$ fontforge name_parser/query_names ../../check-fonts/Terminus/**/*.[ot]tf 2>/dev/null
```


---
_OLD DESCRIPTION:_

**[why]**
When we crate a font we take the OriginalName, add "Nerd Font" and which patches we applied, and add "Mono" if --mono has been specified:

    OriginalName Nerd Font Complete Mono
    OriginalName Nerd Font plus Weather Mono

But the 'Mono' part is quite important, but this scheme will put it in a place where it is easily out of view or has been removed (to keep the name short).

This truncation is especially bad on Windows Compatiple and when the user installs both the 'Nerd Font' and the 'Nerd Font Mono':

    SomeVeryLongFontName Nerd Font Complete
    SomeVeryLongFontName Nerd Font Complete Mono

become after truncation

    SomeVeryLongFontName Nerd Font Comp.ttf
    SomeVeryLongFontName Nerd Font Comp.ttf

**[how]**
Always put the "Mono" directly after "Nerd Font" and all the other name components come later.

#### Requirements / Checklist


#### What does this Pull Request (PR) do?

Change the Family/Full/File name of all `Nerd Font Mono` fonts: Pull the `Mono` directly after the `Nerd Font`

#### How should this be manually tested?

#### Any background context you can provide?

#### What are the relevant tickets (if any)?

#### Screenshots (if appropriate or helpful)
